File: app/src/main/java/develop/y/zhzl/list/presenter/ListPresenterImpl.java
package develop.y.zhzl.list.presenter;

import com.rxnetwork.manager.RxNetWork;

import java.util.List;

import develop.y.zhzl.list.model.ListModel;
import develop.y.zhzl.list.view.IListView;
import framework.api.Api;
import framework.data.Constant;
import framework.mvp.PresenterCompat;

/**
 * by y on 2016/8/7.
 */
public class ListPresenterImpl extends PresenterCompat<List<ListModel>, IListView>
        implements ListPresenter {


    public ListPresenterImpl(IListView mView) {
        super(mView);
    }

    @Override
    protected void onNetWorkNext(List<ListModel> data) {
        getView().removeAllAdapter();
        getView().rxNetWorkSuccess(data);
        getView().isShowEmptyView();
    }

    @Override
    public void netWorkRequest(String suffix, int limit) {
        getView().hideEmptyView();
        RxNetWork.
                getInstance().
                getApi(RxNetWork.observable(Api.ZLService.class).getList(suffix, limit, Constant.OFFSET), this);
    }

}

File: app/src/main/java/develop/y/zhzl/list/presenter/ListPresenterImpl.kt
package develop.y.zhzl.list.presenter

import com.rxnetwork.manager.RxNetWork

import develop.y.zhzl.list.model.ListModel
import develop.y.zhzl.list.view.IListView
import framework.api.Api
import framework.data.Constant
import framework.mvp.PresenterCompat

/**
 * by y on 2016/8/7.
 */
class ListPresenterImpl(mView: IListView) : PresenterCompat<List<ListModel>, IListView>(mView), ListPresenter {

    override fun onNetWorkNext(data: List<ListModel>) {
        view.removeAllAdapter()
        view.rxNetWorkSuccess(data)
        view.isShowEmptyView()
    }

    override fun netWorkRequest(suffix: String, limit: Int) {
        view.hideEmptyView()
        RxNetWork.getInstance().getApi(RxNetWork.observable(Api.ZLService::class.java).getList(suffix, limit, Constant.OFFSET), this)
    }

}
